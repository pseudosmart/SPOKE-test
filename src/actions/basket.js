import { ADD_TO_BASKET, REMOVE_FROM_BASKET, UPDATE_TOTAL } from '../constants/actions'

export const addToBasket = product => dispatch => {
  dispatch({
    type: ADD_TO_BASKET,
    payload: product })
  dispatch({
    type: UPDATE_TOTAL
  })
}

export const removeFromBasket = product => ({
  type: REMOVE_FROM_BASKET,
  payload: product
})



