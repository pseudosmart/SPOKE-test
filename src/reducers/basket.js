import {
  ADD_TO_BASKET,
  REMOVE_FROM_BASKET, //TODO
  UPDATE_TOTAL
} from '../constants/actions'

const intialState = {
  currentBasket: [],
  total: 0
}

export const getCurrentTotal = (currentState) => {
  return currentState.currentBasket.reduce((total, basketItem) => {
    return parseInt(basketItem.price, 10) + total
  }, 0)
}

export default function (currentState = intialState, action) {
  switch (action.type) {
    case ADD_TO_BASKET:
      return {
        ...currentState,
        currentBasket: [...currentState.currentBasket, action.payload]
      }
      case UPDATE_TOTAL:
      return {
        ...currentState,
        total: getCurrentTotal(currentState).toFixed(2)
      }

    default: 
    return currentState
    }
}