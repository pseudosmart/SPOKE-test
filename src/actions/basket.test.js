import * as actions from './basket'

let dispatch

beforeEach(() => {
  dispatch = jest.fn()
})

test('addToBasket', () => {
  expect(actions.addToBasket({ id: 'test', title: 'title' })(dispatch)).toMatchSnapshot()
  expect(dispatch).toHaveBeenCalled()
})

test('removeFromBasket', () => {
  expect(actions.removeFromBasket({ id: 'test', title: 'title' })).toMatchSnapshot()
})
