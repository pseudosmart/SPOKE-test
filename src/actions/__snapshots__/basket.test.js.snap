// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addToBasket 1`] = `undefined`;

exports[`removeFromBasket 1`] = `
Object {
  "payload": Object {
    "id": "test",
    "title": "title",
  },
  "type": "REMOVE_FROM_BASKET",
}
`;
